source/lib-intro.tex

```diff
@@ -1980,189 +2164,388 @@
-  \tcode{true_type} only if an allocator of type \tcode{X} should be swapped
-    when the client container is swapped.
-    See Note B, below.   &
-  \tcode{false_type}        \\ \rowsep
-
-\tcode{X::is_always_equal} &
-  Identical to or derived from \tcode{true_type} or \tcode{false_type}  &
-  \tcode{true_type} only if the expression \tcode{a1 == a2} is guaranteed
-    to be \tcode{true} for any two (possibly \keyword{const}) values
-    \tcode{a1}, \tcode{a2} of type \tcode{X}.   &
-  \tcode{is_empty<X>::\brk{}type}       \\
-
-\end{libreqtab4d}
-
-\pnum
-Note A: The member class template \tcode{rebind} in the table above is
-effectively a typedef template.
-\begin{note}
-In general, if
-the name \tcode{Allocator} is bound to \tcode{SomeAllocator<T>}, then
-\tcode{Allocator::rebind<U>::other} is the same type as
-\tcode{SomeAllocator<U>}, where
-\tcode{SomeAllocator<T>::value_type} is \tcode{T} and
-\tcode{SomeAllocator<U>::\brk{}value_type} is \tcode{U}.
 \end{note}
-If
-\tcode{Allocator} is a class template instantiation of the form
-\tcode{SomeAllocator<T, Args>}, where \tcode{Args} is zero or more type
-arguments, and \tcode{Allocator} does not supply a \tcode{rebind} member
-template, the standard \tcode{allocator_traits} template uses
-\tcode{SomeAllocator<U, Args>} in place of \tcode{Allocator::\brk{}rebind<U>::other}
-by default. For allocator types that are not template instantiations of the
-above form, no default is provided.
 
 \pnum
-Note B:
-If \tcode{X::propagate_on_container_copy_assignment::value} is \tcode{true},
-\tcode{X} shall meet the
-\oldconcept{\-Copy\-Assign\-able} requirements (\tref{cpp17.copyassignable})
-and the copy operation shall not throw exceptions.
-If \tcode{X::propagate_on_container_move_assignment::value} is \tcode{true},
-\tcode{X} shall meet the
-\oldconcept{\-Move\-Assign\-able} requirements (\tref{cpp17.moveassignable})
-and the move operation shall not throw exceptions.
-If \tcode{X::propagate_on_container_swap::value} is \tcode{true},
-lvalues of type \tcode{X} shall be swappable\iref{swappable.requirements}
-and the \tcode{swap} operation shall not throw exceptions.
+\remarks
+If \tcode{n == 0}, the return value is unspecified.
+\end{itemdescr}
+
+\begin{itemdecl}
+a.allocate(n, y)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{X::pointer}
+
+\pnum
+\effects
+Same as \tcode{a.allocate(n)}.
+The use of \tcode{y} is unspecified, but it is intended as an aid to locality.
+
+\pnum
+\remarks
+Default: \tcode{a.allocate(n)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a.allocate_at_least(n)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+allocation_result<X::pointer>
 
 \pnum
-Note C:
+\returns
+\tcode{allocation_result<X::pointer>\{ptr, count\}}
+where \tcode{ptr} is memory allocated for an array of \tcode{count} \tcode{T}
+and such an object is created but array elements are not constructed,
+such that $\tcode{count} \geq \tcode{n}$.
 If \tcode{n == 0}, the return value is unspecified.
```

Comment:
This is a _Remarks_ in `a.allocate(n)`.

Reply:
Yes, but (pre-existing), `a.allocate(n)` doesn't even say what it returns in the good case, so I can't put it in the right spot.

Reply:
Makes sense.